orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/CreateServerGroupTaskSpec.groovy

```diff
@@ -76,4 +82,484 @@ class CreateServerGroupTaskSpec extends Specification {
+    deployStage.context.deploymentDetails = [
+      ["imageId": "not-my-ami", "ami": "not-my-ami", "region": deployRegion],
+      ["imageId": "also-not-my-ami", "ami": "also-not-my-ami", "region": deployRegion]
+    ]
+    makeDependentOn(deployStage, bakeStage1)
+
+    def deployTask = new CreateServerGroupTask(kato: katoService, serverGroupCreators: buildServerGroupCreators(mortService))
+
+    when:
+    def result = deployTask.execute(deployStage)
+
+    then:
+    _ * mortService.getAccountDetails("abc") >> [:]
+    1 * katoService.requestOperations(operationCloudProvider, {
+      return it[createOpIndex]?.createServerGroup?.get(imageAttributeKey) == expectedImageId
+    }) >> { Observable.from(taskId) }
+    // This helps avoid an NPE within CreateServerGroupTask; this results in better error-reporting on a test failure.
+    _ * katoService.requestOperations(operationCloudProvider, _) >> { Observable.from(taskId) }
+    result?.stageOutputs == baseOutput + ["kato.result.expected": katoResultExpected]
+
+    where:
+    operationCloudProvider | bakeCloudProvider | createOpIndex | imageAttributeKey | katoResultExpected || expectedImageId
+    "gce"                  | "gce"             | 0             | "image"           | false              || "some-ami-name"
+    "gce"                  | null              | 0             | "image"           | false              || "some-ami-name"
+    "aws"                  | "aws"             | 1             | "imageId"         | true               || "some-ami-name"
+    "aws"                  | null              | 1             | "imageId"         | true               || "some-ami-name"
+  }
+
+  @Unroll
+  def "image for #bakeCloudProvider in parent pipeline deployment details is resolved by create stage for #operationCloudProvider"() {
+    /*
+      parentPipeline: deploymentDetails
+                                        | trigger
+                                        v
+                                          childPipeline: manualJudgment -> createServerGroup
+     */
+    given:
+    KatoService katoService = Mock(KatoService)
+    MortService mortService = Mock(MortService)
+    def deployRegion = "us-east-1"
+    def parentGlobalContext = [
+      deploymentDetails: [
+        ["imageId": "parent-ami", "ami": "parent-ami", "region": deployRegion]
+      ]
+    ]
+    def parentPipeline = new Pipeline.Builder().withName("parent").withGlobalContext(parentGlobalContext).build()
+
+    def childTrigger = [
+      parentExecution: parentPipeline
+    ]
+    def childPipeline = new Pipeline.Builder().withName("child").withTrigger(childTrigger).build()
+    def manualJudgmentStage = buildStageForPipeline(childPipeline, "manualJudgment")
+
+    def deployConfig = buildDeployConfig(deployRegion, operationCloudProvider)
+    def deployStage = buildStageForPipeline(childPipeline, "createServerGroup", deployConfig)
+    makeDependentOn(deployStage, manualJudgmentStage)
```

Comment:
FWIW ... @robfletcher and I chatted and he's going to come up with a little builder that makes it easier to stitch together pipelines and hierarchical children, etc. 

We do this all to much.